src/azure-cli/azure/cli/__main__.py

```diff
@@ -61,15 +62,12 @@ def cli_main(cli, args):
     raise ex
 
 finally:
-    try:
-        # Log the invoke finish time
-        invoke_finish_time = timeit.default_timer()
-        logger.info("Command ran in %.3f seconds (init: %.3f, invoke: %.3f)",
-                    invoke_finish_time - start_time,
-                    init_finish_time - start_time,
-                    invoke_finish_time - init_finish_time)
-    except NameError:
```

Comment:
`NameError` is introduced in https://github.com/adewaleo/azure-cli/blob/26a3ddf7330b2958ed0fa40cfb74cd8477013ff7/src/azure-cli/azure/cli/__main__.py#L58-L62.
I think `NameError` is not raised unless `telemetry.start()` fails.

Reply:
Indeed, `NameError` should never be **caught and handled** by us. Instead, it should be **avoided** in the code itself.